psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,799 @@
+        },
+        5: {
+            "sym": ((-2, ), (-1, ), (1, ), (2, )),
+            "asym": ((-2, ), (-1, )),
+            "off": ((-1, -2), (-2, -1), (-1, -1), (1, -1), (-1, 1), (1, 1), (2, 1), (1, 2))
+        }
+    }
+
+    # We won't need disps for a while, but let's validate num_pts early.
+    try:
+        disps = pts_dict[num_pts]
+    except KeyError:
+        raise ValidationError("FINDIF: Invalid number of points!")
+
+    # Convention: x_pi means x_per_irrep. The ith element is x for irrep i, with Cotton ordering.
+    salc_indices_pi = [[] for h in range(Nirrep)]
+
+    # Validate that we have an irrep matching the user-specified irrep, if any.
+    try:
+        salc_indices_pi[freq_irrep_only]
+    except (TypeError, IndexError):
+        if freq_irrep_only != -1:
+            raise ValidationError("FINDIF: Irrep value not in valid range.")
+
+    # Populate salc_indices_pi for all irreps.
+    for i, salc in enumerate(salc_list):
+        salc_indices_pi[salc.irrep_index()].append(i)
+
+    # If the method allows more than one irrep, print how the irreps partition the SALCS.
+    if print_lvl and method_allowed_irreps != 0x1 and verbose:
+        core.print_out("\tIndex of SALCs per irrep:\n")
+        for h in range(Nirrep):
+            if print_lvl > 1 or freq_irrep_only in {h, -1}:
+                tmp = (" {:d} " * len(salc_indices_pi[h])).format(*salc_indices_pi[h])
+                core.print_out("\t {:d} : ".format(h + 1) + tmp + "\n")
+        core.print_out("\tNumber of SALCs per irrep:\n")
+        for h in range(Nirrep):
+            if print_lvl > 1 or freq_irrep_only in {h, -1}:
+                core.print_out("\t Irrep {:d}: {:d}\n".format(h + 1, len(salc_indices_pi[h])))
+
+    # Now that we've printed the SALCs, clear any that are not of user-specified symmetry.
+    if freq_irrep_only != -1:
+        for h in range(Nirrep):
+            if h != freq_irrep_only:
+                salc_indices_pi[h].clear()
+
+    Ndisp_pi = []
+
+    for irrep, indices in enumerate(salc_indices_pi):
+        Ndisp = len(indices) * len(disps["asym" if irrep != 0 else "sym"])
+        if mode == "2_0":
+            # Either len(indices) or len(indices)-1 is even, so dividing by two is safe.
+            Ndisp += len(indices) * (len(indices) - 1) // 2 * len(disps["off"])
+        Ndisp_pi.append(Ndisp)
+
+    # Let's print out the number of geometries, the displacement multiplicity, and the CdSALCs!
+    if print_lvl and verbose:
+        core.print_out("\tNumber of geometries (including reference) is {:d}.\n".format(sum(Ndisp_pi) + 1))
+        if method_allowed_irreps != 0x1:
+            core.print_out("\tNumber of displacements per irrep:\n")
+            for i, ndisp in enumerate(Ndisp_pi, start=1):
+                core.print_out("\t  Irrep {:d}: {:d}\n".format(i, ndisp))
+
+    if print_lvl > 1 and verbose:
+        for salc in salc_list:
+            salc.print_out()
+
+    data.update({
+        "Ndisp_pi": Ndisp_pi,
```

Comment:
Mixing cases here a bit, for pure snake case this should be like `ndisp_pi` or `n_disp_pi`.